### ISSUE TITLE
Pylint crash while pre-commit

### ISSUE THREAD

**Issue by shmir** [pylint-dev/pylint#9240]
[pylint-crash-2023-11-16-14-50-53.txt](https://github.com/pylint-dev/pylint/files/13377896/pylint-crash-2023-11-16-14-50-53.txt)

This is the pre-commit config:
  - repo: local
    hooks:
      - id: pylint
        name: pylint
        entry: pylint
        language: system
        types: [python]
        args: [
          --max-line-length=127,
          '--disable=logging-fstring-interpolation,logging-not-lazy,unspecified-encoding,logging-format-interpolation',
          '--good-names=_,rc,ti,ip,l3',
          '--load-plugins=pylint_pytest'
        ]

Python version is 12.

**Comment by jacobtylerwalls** [comment]
Can you provide the output of `pylint --version`? Python 3.12 is only supported with pylint 3.0 and astroid 3.0.

**Comment by jacobtylerwalls** [comment]
Duplicate of #9091

**Comment by shmir** [comment]
After removing pylint-pytest and upgrading pylint as foll0wing:

(venv12) ~/PycharmProjects/multimedia (yoram-production) $ pip show pylint
Name: pylint
Version: 3.0.2
Summary: python code static checker
Home-page: 
Author: 
Author-email: Python Code Quality Authority <code-quality@python.org>
License: GPL-2.0-or-later
Location: /home/yorams/PycharmProjects/multimedia/venv12/lib/python3.12/site-packages
Requires: astroid, dill, dill, isort, mccabe, platformdirs, tomlkit
Required-by: 
(venv12) ~/PycharmProjects/multimedia (yoram-production) $ pip show astroid
Name: astroid
Version: 3.0.1
Summary: An abstract syntax tree for Python with inference support.
Home-page: 
Author: 
Author-email: 
License: LGPL-2.1-or-later
Location: /home/yorams/PycharmProjects/multimedia/venv12/lib/python3.12/site-packages
Requires: 
Required-by: pylint
(venv12) ~/PycharmProjects/multimedia (yoram-production) $ 

It seems to be working.
Thanks.
